tests/utils.py

```diff
@@ -330,9 +328,17 @@ def file_content(path: Path) -> str | None:
 
 def assert_url_resolvable(url: str):
     """Tests whether a URL is resolvable with retries; raises for failure."""
-    with requests.session() as sess:
-        adapter = HTTPAdapter(max_retries=Retry(status_forcelist={500, 502, 504}))
-        sess.mount("http://", adapter)
-        sess.mount("https://", adapter)
-
-        sess.head(url).raise_for_status()
+    transport = httpx.HTTPTransport(
```

Comment:
httpx has no equivalent to urllib3's Retry and the requests adapter pattern.

Generally speaking, httpx's transports replace the adapter pattern, and in this case, they handle retrying in the case of network connectivity issues.

There is nothing like `Retry` and its `status_forcelist`, so the while loop reimplements that aspect.

I chose to add 3 retries somewhat randomly. `Retry` uses 10 retries by default, so I could increase `retries` and `bad_response_retries` to 5 to match, but the even split between networking errors and status code errors is arbitrary. requests does not operate that way, and there's currently no way to implement a combined retry count in httpx without overriding private methods deep in httpcore's connection handling.